api/types/resource_ids.go

```diff
@@ -83,6 +85,78 @@ func (id *ResourceID) validateK8sSubResource() error {
 	return nil
 }
 
+func (idl *ResourceAccessIDList) CheckAndSetDefaults() error {
+	for _, r := range idl.Resources {
+		rid, rc := r.GetResourceID(), r.GetConstraints()
+		if err := rid.CheckAndSetDefaults(); err != nil {
+			return trace.Wrap(err)
+		}
+		if rc == nil {
+			continue
+		} else if err := rc.CheckAndSetDefaults(); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+func (r *ResourceAccessID) GetResourceID() ResourceID {
+	return r.Id
+}
+
+func (r *ResourceAccessID) GetConstraints() *ResourceConstraints {
+	return r.Constraints
+}
+
+// ResourceIDsToResourceAccessIDs wraps a slice of [ResourceID]s into
+// [ResourceAccessID] instances with no additional fields populated.
+func ResourceIDsToResourceAccessIDs(ids []ResourceID) []ResourceAccessID {
+	wrapped := make([]ResourceAccessID, 0, len(ids))
+	for _, id := range ids {
+		wrapped = append(wrapped, ResourceAccessID{Id: id})
+	}
+	return wrapped
+}
+
+// CombineAsResourceAccessIDs converts plain [ResourceID]s to [ResourceAccessID]s
+// and combines them with existing [ResourceAccessID]s into a single slice.
+func CombineAsResourceAccessIDs(ids []ResourceID, accessIDs []ResourceAccessID) []ResourceAccessID {
+	totalLen := len(ids) + len(accessIDs)
+	zipped := make([]ResourceAccessID, 0, totalLen)
+	for _, id := range ids {
+		zipped = append(zipped, ResourceAccessID{Id: id})
+	}
```

Comment:
Consider reusing `ResourceIDsToResourceAccessIDs`. This will protect from modifying one and not another if the semantics of conversion change at any point.